lib/Frontend/Frontend.cpp

```diff
@@ -497,58 +565,131 @@ void CompilerInstance::performSema() {
   if (options.DebugTimeExpressionTypeChecking) {
     TypeCheckOptions |= TypeCheckingFlags::DebugTimeExpressions;
   }
+  return TypeCheckOptions;
+}
+
+// Return true if had load error
+bool CompilerInstance::parsePartialModulesAndLibraryFiles(
+    SourceFile::ImplicitModuleImportKind implicitModuleImportKind,
+    ModuleDecl *objCModuleUnderlyingMixedFramework,
+    ModuleDecl *importedHeaderModule,
+    SmallVectorImpl<ModuleDecl *> &importModules,
+    PersistentParserState &PersistentState,
+    DelayedParsingCallbacks *DelayedParseCB) {
+  SharedTimer timer("performSema-parsePartialModulesAndLibraryFiles");
+  bool hadLoadError = false;
+  // Parse all the partial modules first.
+  for (auto &PM : PartialModules) {
+    assert(PM.ModuleBuffer);
+    if (!SML->loadAST(*MainModule, SourceLoc(), std::move(PM.ModuleBuffer),
+                      std::move(PM.ModuleDocBuffer)))
+      hadLoadError = true;
+  }
+
+  // Then parse all the library files.
+  for (auto BufferID : BufferIDs) {
+    if (BufferID != MainBufferID)
+      parseALibraryFile(BufferID, implicitModuleImportKind,
```

Comment:
Nitpick: please put braces around multi-*line* `if` bodies even if it's only a single *statement.*

Reply:
Weird--it *is* a single line on my screen:

 // Then parse all the library files.
  for (auto BufferID : BufferIDs) {
    if (BufferID != MainBufferID)
      parseALibraryFile(BufferID, implicitImports, PersistentState, DelayedParseCB);
  }

Do they get rewrapped somehow?

But I'm happy to add the braces and have done so.

Reply:
Oops, GitHub's wrapping tricked me. If it fits in [80 columns](http://llvm.org/docs/CodingStandards.html#source-code-width), it's fine.